### PR TITLE
adding LOG and exception handling in the clean cache

### PR DESCRIPTION
**Pull request by miranov25**
Hello @sawenzel 

These changes solved the cache clean-up problems I described in an earlier email. 
It was important to include their try-catch in the file listing

The logging is too verbose, I kept it verbose to demonstrate the problem. Feel free to remove the part that is not necessary.

### CODE REVIEW

File: Common/Utils/src/FileSystemUtils.cxx [hunk 1]
```diff
@@ -32,12 +34,17 @@ std::vector<std::string> listFiles(std::string const& dir, std::string const& se
...
-      }
-    }
+    try {
+    	if (!p.is_directory()) {
+     		 auto fn = p.path().filename().string();
+      		if (regex_match(fn, str_expr)) {
+        	filenames.push_back(p.path().string());
+      	}
+    	}
+    }catch(...){
```
Comment (sawenzel) [inline Common/Utils/src/FileSystemUtils.cxx:44]: The approach looks good to me. But I think we can do without logging everything.
So could you please remove all the LOG(info) statements included?

Comment (miranov25) [reply Common/Utils/src/FileSystemUtils.cxx:44]: Yes. I will remove soon

Comment (miranov25) [reply Common/Utils/src/FileSystemUtils.cxx:44]: Is it better now?
I keep LOG(warn) and LOG - begin/end

File: Common/Utils/src/FileSystemUtils.cxx [hunk 2]
```diff
@@ -18,6 +18,8 @@
 #include <vector>
 #include <regex>
 #include <iostream>
+#include "FairLogger.h"
```
Comment (sawenzel) [inline Common/Utils/src/FileSystemUtils.cxx:21]: take away

File: Common/Utils/src/FileSystemUtils.cxx [hunk 3]
```diff
@@ -32,12 +34,16 @@ std::vector<std::string> listFiles(std::string const& dir, std::string const& se
...
+    try {
+    	if (!p.is_directory()) {
+     		 auto fn = p.path().filename().string();
+      		if (regex_match(fn, str_expr)) {
+        	filenames.push_back(p.path().string());
+      	}
+    	}
+    }catch(...){
+      continue;
+   }
```
Comment (sawenzel) [inline Common/Utils/src/FileSystemUtils.cxx:46]: please fix clang formatting PR

File: run/o2sim_parallel.cxx [hunk 4]
```diff
@@ -71,13 +71,22 @@ void remove_tmp_files()
 {
   // remove all (known) socket files in /tmp
   // using the naming convention /tmp/o2sim-.*PID
+  //printf("remove_tmp_files\n");     
+  LOG(info) << "remove_tmp_files: BEGIN ";
```
Comment (sawenzel) [inline run/o2sim_parallel.cxx:75]: please take away. I prefer not to have more printouts.

File: run/o2sim_parallel.cxx [hunk 5]
```diff
@@ -71,13 +71,22 @@ void remove_tmp_files()
...
-    std::filesystem::remove(std::filesystem::path(fn));
+    continue;
+    try{
+    	std::filesystem::remove(std::filesystem::path(fn));
+    }
+    catch(...){
+        LOG(warn) << "Invalid file " << fn << " from /tmp ";
+    }
   }
+  LOG(info) << "remove_tmp_files: END ";
```
Comment (sawenzel) [inline run/o2sim_parallel.cxx:89]: same here.

**Comment by miranov25** [comment]
Hello @sawenzel 

I have not set up an environment for the Coding Convention yet, and I am spending the next few days working on the code for the machine learning workshop. So I will not update until next week. Could you make the change you suggested or wait another week?

Regards
Marian

**Comment by sawenzel** [comment]
Ok. I'll do it myself.

**Comment by miranov25** [comment]
Redirected to #8517

**Pull request closed without merging**